pkg/apis/common/v1/conf.go

```diff
@@ -60,10 +59,8 @@ func extractAssocConfFromAnnotation(annotations map[string]string, annotationNam
 }
 
 func unsafeStringToBytes(s string) []byte {
-	hdr := *(*reflect.StringHeader)(unsafe.Pointer(&s))    //nolint:govet
-	return *(*[]byte)(unsafe.Pointer(&reflect.SliceHeader{ //nolint:govet
-		Data: hdr.Data,
-		Len:  hdr.Len,
-		Cap:  hdr.Len,
-	}))
+	if s == "" {
+		return nil
+	}
+	return unsafe.Slice(unsafe.StringData(s), len(s))
```

Comment:
I am confused as to why we need `unsafe` here at all. I think I need a bit of explanation why can't we simply do `[]byte(s)`? https://go.dev/play/p/Z_vrBU8UGe0

Reply:
I accidentally deleted the comment in #5489.

```
// unsafeStringToBytes converts a string to a byte array without making extra allocations.
// since we read potentially large strings from annotations on every reconcile loop, this should help
// reduce the amount of garbage created.
```

Reply:
Maybe still a premature optimisation? But I am OK to keep as it was this way before.

Reply:
Yeah, that seems like a premature optimization. It's terrible because once in place, it becomes uncomfortable to remove it without thoroughly evaluating its real impact. I didn't dare question it.